exercises/formatSeconds/formatSeconds.js

```diff
@@ -17,10 +17,32 @@
  */
 
 function formatSeconds(num) {
-  // This is your job. :)
+  
+  // Create reference dictionary for lengths of time in seconds
+  const secondsRef = {'w':604800, 'd':86400, 'h':3600, 'm':60, 's':1};
+  
+  // Create empty array to push formatted lengths of time
+  const formattedList = [];
+  
+  // Define a working variable for seconds remaining
+  let secondsRemaining = num;
+  
+  // Use for loop to push formatted lengths of time to the array
+  for (let key of Object.keys(secondsRef)){
+    let amount = Math.floor(secondsRemaining/secondsRef[key]);
+    formattedList.push(amount += key);   
```

Comment:
Consider:

```javascript
if (amount > 0 || formattedList.length > 0) {
  formattedList.push(amount + key);
}
```

Then there will be no need for a separate loop to remove the leading elements.

Reply:
Also, the `+=` here isn't what you want. You just want to concatenate the two; not change the value of `amount`.

I also recommend explicitly converting to strings rather than relying on JavaScript's weak typing, e.g.,

```javascript
formattedLish.push(String(amount) + key);
```

Relying on JS's implicit type conversions obscures the programmer's intention and will bite you at some point.

A classic (and funny) talk on the subject: https://www.destroyallsoftware.com/talks/wat